samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -134,6 +136,24 @@ public LocalApplicationRunner(Config config) {
   }
 
   @Override
+  public void runTask() {
```

Comment:
we discussed it, and decided that this is a quick fix for people to try it with least amount of changes, and this API will be hidden with Yi's API changes later. And then it will most likely change to accept a factory, as you described.